### ISSUE TITLE
How to use SigLIP in training?

### ISSUE THREAD

**Issue by mtulgac** [Simon-Lepage/CondViT-LRVSF#1]
The model uses ViT-B/16 as CLIP backbone for image representation, and this could be changed with other CLIP models with minor changes in ConditionalViT class. 

I was wondering if we can also use SigLIP (and SigLIP 2 now), and if so, what are the changes that must be done?

**Comment by Simon-Lepage** [comment]
I only experimented with CLIP and ViT-B/32 & ViT-B/16 architectures, so I'm not 100% sure, but of course feel free to try.

Off the top of my head :
- Clone and modify SigLIP classes to include the conditioning token and its positional encoding : as learnable parameters in the `__init__` and as arguments in `forward`
- Do something with the temperature, the `logit_scale` attribute of the CLIP model, I don't know how it's handled by SigLIP but it's used multiple times in this repo.
- Maybe modify the data normalization to fit SigLIP's pretraining ?
- I think SigLIP uses average pooling instead of a CLS token for the final representation, I don't know how much of a difference it makes. 

And of course some boilerplate code (I probably forget some of it) : 
- Modify the utils to load/initialize your model from pretrained SigLIP despite the new parameters (see the `ckpt_utils.py` file)
- Change the argument parsing in `main.py`

Good luck

**Comment by mtulgac** [comment]
Thank you for your response.
